File: livingdoc-api/src/main/java/org/livingdoc/api/fixtures/scenarios/Binding.java
package org.livingdoc.api.fixtures.scenarios;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;


/**
 * Binds a name to the annotated parameter. If the bound name occurs as parameter in a {@link Step}
 * annotation of this method, the extracted value will be converted and passed to this method during
 * execution.
 *
 * @see Step
 * @see ScenarioFixture
 * @since 2.0
 */
@Target(ElementType.PARAMETER)
@Retention(RetentionPolicy.RUNTIME)
public @interface Binding {
    String value();
}

File: livingdoc-api/src/main/kotlin/org/livingdoc/api/fixtures/scenarios/Binding.kt
package org.livingdoc.api.fixtures.scenarios

/**
 * Binds a name to the annotated parameter. If the bound name occurs as parameter in a [Step] annotation of this method,
 * the extracted value will be converted and passed to this method during execution.
 *
 * @see Step
 * @see ScenarioFixture
 *
 * @since 2.0
 */
@Target(
    AnnotationTarget.PROPERTY,
    AnnotationTarget.FIELD,
    AnnotationTarget.VALUE_PARAMETER
)
annotation class Binding(val value: String)
